RecoTracker/LSTCore/src/alpaka/EndcapGeometry.dev.cc

```diff
@@ -62,24 +72,18 @@ void SDL::EndcapGeometry<SDL::Dev>::fillGeoMapArraysExplicit(SDL::QueueAcc& queu
   }
 
   nEndCapMap = counter;
-
-  // Copy data from host to device buffers
-  alpaka::memcpy(queue, geoMapPhi_buf, mapPhi_host_buf);
-  alpaka::memcpy(queue, geoMapDetId_buf, mapDetId_host_buf);
-  alpaka::wait(queue);
 }
 
-float SDL::EndcapGeometry<SDL::Dev>::getdxdy_slope(unsigned int detid) const {
-  if (dxdy_slope_.find(detid) != dxdy_slope_.end()) {
-    return dxdy_slope_.at(detid);
-  } else {
-    return 0;
-  }
-}
-float SDL::EndcapGeometryHost<SDL::Dev>::getdxdy_slope(unsigned int detid) const {
+template <typename TDev>
+float SDL::EndcapGeometry<TDev>::getdxdy_slope(unsigned int detid) const {
   if (dxdy_slope_.find(detid) != dxdy_slope_.end()) {
     return dxdy_slope_.at(detid);
   } else {
     return 0;
   }
 }
+
+// We need to instantiate these to ensure that the compiler generates them
+template class SDL::EndcapGeometry<SDL::DevHost>;
```

Comment:
doesn't this lead to the same and visible symbols in each of the backend?
It's perhaps better to make a host level `EndcapGeometry` which doesn't have the buffers and on the alpaka side have only the `EndcapGeometryBuffers`.
IIUC, the current implementation of `EndcapGeometry` by construction does not allow meaningful access to `::load` or `::getdxdy_slope`.

Reply:
Yeah, that's a good idea. It might also be a good opportunity to start moving things out of the `alpaka` directory. I'll see how that goes.

Reply:
I made the separation you suggested. Now `EndcapGeometry` is analogous to `TiltedGeometry` and `EndcapGeometryBuffer` is analogous to `moduleBuffer`. I didn't move things out of the `alpaka` directory yet, but I'll start thinking about that.